### PR TITLE
Prepare v6.0.0 release (1st PS 0.15.0-compatible release)

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Backlinking to purescript/purescript#4244. Prepares project for first release that is compatible with PureScript v0.15.0.

:ping_pong: @thomashoneyman

:robot: This is an automated pull request to prepare the next release of this library. PR was created via the [Release.purs](https://github.com/JordanMartinez/purescript-ecosystem-update/blob/master/src/purescript/Command/Release.purs) file. Some of the following steps are already done; others should be performed by a human once the pull request is merged:

- [x] Updated bower dependencies to 0.15.0-compatible versions
- [x] ci.yml: Node updated to 14.
- [x] ci.yml: No spago tests to uncomment.
- [x] Updated changelog
- [ ] Publish a GitHub [release](https://github.com/purescript/purescript-foldable-traversable/releases/new?tag=v6.0.0&title=v6.0.0&body=Breaking%20changes%3A%0A-%20Migrate%20FFI%20to%20ES%20modules%20%28%23146%20by%20%40kl0tl%20and%20%40JordanMartinez%29%0A-%20Drop%20deprecated%20%60foldMap1Default%60%20%28%23147%20by%20%40JordanMartinez%29%0A%0ANew%20features%3A%0A%0ABugfixes%3A%0A%0AOther%20improvements%3A%0A-%20Narrow%20down%20unnecessarily%20imprecise%20type%20of%20%60mapWithIndexArray%60%20%28%23145%29
).
- [ ] Upload the release to Pursuit with `pulp publish`.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -17,23 +17,23 @@
...
+    "purescript-tuples": "^7.0.0"
   },
   "devDependencies": {
-    "purescript-assert": "master",
-    "purescript-console": "master",
-    "purescript-integers": "master",
+    "purescript-assert": "^6.0.0",
+    "purescript-console": "^6.0.0",
+    "purescript-integers": "^6.0.0",
     "purescript-math": "master",
```
Comment (thomashoneyman) [inline bower.json:36]:
```suggestion
```

**Review by thomashoneyman — APPROVED** [review]

**Comment by thomashoneyman** [comment]
Uh oh! Gotta update the tests.

**Comment by JordanMartinez** [comment]
Latest commit should fix the test, but I haven't tested locally.

**Comment by JordanMartinez** [comment]
Tests pass now.

**Pull request merged**